pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogic_synthetic_privileges failed

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_synthetic_privileges [failed](https://mesolite.cluster.engflow.com/invocations/default/b333972b-b412-438b-82a9-f63f1e3f71ef?testReportRun=1&testReportShard=38&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy8zbm9kZS10ZW5hbnQ6M25vZGUtdGVuYW50X3Rlc3Q=) on release-24.3 @ [1e1cfc7722be3953f5eff7b92ab53b1c811ab9b1](https://github.com/cockroachdb/cockroach/commits/1e1cfc7722be3953f5eff7b92ab53b1c811ab9b1):


```
=== RUN   TestTenantLogic_synthetic_privileges
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestTenantLogic_synthetic_privileges2099809142
    test_log_scope.go:76: use -show-logs to present logs inline
[17:07:25] setting distsql_workmem='81644B';
[17:07:25] setting distsql_workmem='81644B';
[17:07:27] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 62 statements
[17:07:34] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 68 statements
    logic.go:3045: 
         
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:307: SELECT has_table_privilege('testuser4', 'crdb_internal.feature_usage', 'SELECT')
        expected success, but found
        (40001) restart transaction: has_table_privilege(): TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=bdce9548 key=/Tenant/10/Table/52/1/"testuser4"/"/vtable/crdb_internal/feature_usage" iso=Serializable pri=100.00000000 epo=0 ts=1737738447.891715135,0 min=1737738447.891715135,0 seq=4} lock=true stat=PENDING rts=1737738447.891715135,0 wto=false gul=1737738448.391715135,0
        eval.go:2172: in MaybeWrapError()
[17:07:34] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges with config 3node-tenant: 69 tests, 1 failures
    logic.go:4231: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:312: error while processing
    logic.go:4231: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:312: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantLogic_synthetic_privileges2099809142
--- FAIL: TestTenantLogic_synthetic_privileges (10.40s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=38</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_synthetic_privileges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-46801

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_synthetic_privileges [failed](https://tanzanite.cluster.engflow.com/invocations/default/47d4174f-92c7-467f-994e-0861e87db3d7?testReportRun=22&testReportShard=38&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy8zbm9kZS10ZW5hbnQ6M25vZGUtdGVuYW50X3Rlc3Q=) on release-24.3 @ [3bb549350582192ffa091bfecba15527e545e30b](https://github.com/cockroachdb/cockroach/commits/3bb549350582192ffa091bfecba15527e545e30b):


```
=== RUN   TestTenantLogic_synthetic_privileges
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestTenantLogic_synthetic_privileges539431191
    test_log_scope.go:76: use -show-logs to present logs inline
[07:07:44] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 65 statements
[07:07:49] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 68 statements
    logic.go:3045: 
         
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:307: SELECT has_table_privilege('testuser4', 'crdb_internal.feature_usage', 'SELECT')
        expected success, but found
        (40001) restart transaction: has_table_privilege(): TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=b3d8f24c key=/Tenant/10/Table/52/1/"testuser4"/"/vtable/crdb_internal/feature_usage" iso=Serializable pri=100.00000000 epo=0 ts=1737961664.265612853,0 min=1737961664.265612853,0 seq=4} lock=true stat=PENDING rts=1737961664.265612853,0 wto=false gul=1737961664.765612853,0
        eval.go:2172: in MaybeWrapError()
[07:07:49] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges with config 3node-tenant: 69 tests, 1 failures
    logic.go:4231: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:312: error while processing
    logic.go:4231: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:312: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantLogic_synthetic_privileges539431191
--- FAIL: TestTenantLogic_synthetic_privileges (9.32s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=22</code>
 - <code>shard=38</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139890 pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogic_synthetic_privileges failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_synthetic_privileges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_synthetic_privileges [failed](https://tanzanite.cluster.engflow.com/invocations/default/916d1857-e312-4220-bea7-b8002a68246b?testReportRun=23&testReportShard=38&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy8zbm9kZS10ZW5hbnQ6M25vZGUtdGVuYW50X3Rlc3Q=) on release-24.3 @ [eaa38b5fe50c68d6de15a17b984d937c8af6f94d](https://github.com/cockroachdb/cockroach/commits/eaa38b5fe50c68d6de15a17b984d937c8af6f94d):


```
=== RUN   TestTenantLogic_synthetic_privileges
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestTenantLogic_synthetic_privileges3020172229
    test_log_scope.go:76: use -show-logs to present logs inline
[08:37:02] setting distsql_workmem='24282B';
[08:37:02] setting distsql_workmem='24282B';
[08:37:04] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 62 statements
    logic.go:3045: 
         
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:307: SELECT has_table_privilege('testuser4', 'crdb_internal.feature_usage', 'SELECT')
        expected success, but found
        (40001) restart transaction: has_table_privilege(): TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=8570cd71 key=/Tenant/10/Table/52/1/"testuser4"/"/vtable/crdb_internal/feature_usage" iso=Serializable pri=100.00000000 epo=0 ts=1738139824.949758238,0 min=1738139824.949758238,0 seq=4} lock=true stat=PENDING rts=1738139824.949758238,0 wto=false gul=1738139825.449758238,0
        eval.go:2172: in MaybeWrapError()
[08:37:11] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 69 statements
[08:37:11] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges with config 3node-tenant: 69 tests, 1 failures
    logic.go:4231: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:312: error while processing
    logic.go:4231: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:312: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantLogic_synthetic_privileges3020172229
--- FAIL: TestTenantLogic_synthetic_privileges (10.32s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=23</code>
 - <code>shard=38</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #140011 pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogic_synthetic_privileges failed [C-test-failure O-robot P-2 T-sql-foundations X-duplicate branch-release-24.3.4-rc]
- #139890 pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogic_synthetic_privileges failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_synthetic_privileges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_synthetic_privileges [failed](https://tanzanite.cluster.engflow.com/invocations/default/86468f42-a0c6-438b-9cc2-26c5d83adc16?testReportRun=12&testReportShard=38&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy8zbm9kZS10ZW5hbnQ6M25vZGUtdGVuYW50X3Rlc3Q=) on release-24.3 @ [f7e416c59805f2478865884a031b5a4eb9e7c8cb](https://github.com/cockroachdb/cockroach/commits/f7e416c59805f2478865884a031b5a4eb9e7c8cb):


```
=== RUN   TestTenantLogic_synthetic_privileges
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestTenantLogic_synthetic_privileges1287811902
    test_log_scope.go:76: use -show-logs to present logs inline
[19:45:41] setting distsql_workmem='96768B';
[19:45:41] setting distsql_workmem='96768B';
[19:45:43] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 60 statements
    logic.go:3045: 
         
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:307: SELECT has_table_privilege('testuser4', 'crdb_internal.feature_usage', 'SELECT')
        expected success, but found
        (40001) restart transaction: has_table_privilege(): TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_ABORT_SPAN): "sql txn" meta={id=299a25c2 key=/Tenant/10/Table/52/1/"testuser4"/"/vtable/crdb_internal/feature_usage" iso=Serializable pri=100.00000000 epo=0 ts=1738266343.710770158,0 min=1738266343.710770158,0 seq=4} lock=true stat=ABORTED rts=1738266343.710770158,0 wto=false gul=1738266344.210770158,0
        eval.go:2172: in MaybeWrapError()
[19:45:49] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 69 statements
[19:45:49] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges with config 3node-tenant: 69 tests, 1 failures
    logic.go:4231: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:312: error while processing
    logic.go:4231: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:312: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantLogic_synthetic_privileges1287811902
--- FAIL: TestTenantLogic_synthetic_privileges (10.50s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=12</code>
 - <code>shard=38</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #140011 pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogic_synthetic_privileges failed [C-test-failure O-robot P-2 T-sql-foundations X-duplicate branch-release-24.3.4-rc]
- #139890 pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogic_synthetic_privileges failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_synthetic_privileges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_synthetic_privileges [failed](https://tanzanite.cluster.engflow.com/invocations/default/90655a5e-a03d-41e7-b858-daa377e923e7?testReportRun=6&testReportShard=38&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy8zbm9kZS10ZW5hbnQ6M25vZGUtdGVuYW50X3Rlc3Q=) on release-24.3 @ [af56876f2593fbc5c763c3dc389fae31d19de688](https://github.com/cockroachdb/cockroach/commits/af56876f2593fbc5c763c3dc389fae31d19de688):


```
=== RUN   TestTenantLogic_synthetic_privileges
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestTenantLogic_synthetic_privileges1662421058
    test_log_scope.go:76: use -show-logs to present logs inline
[16:12:13] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 60 statements
[16:12:19] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 68 statements
    logic.go:3045: 
         
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:307: SELECT has_table_privilege('testuser4', 'crdb_internal.feature_usage', 'SELECT')
        expected success, but found
        (40001) restart transaction: has_table_privilege(): TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=d58101df key=/Tenant/10/Table/52/1/"testuser4"/"/vtable/crdb_internal/feature_usage" iso=Serializable pri=100.00000000 epo=0 ts=1738685533.717710037,0 min=1738685533.717710037,0 seq=4} lock=true stat=PENDING rts=1738685533.717710037,0 wto=false gul=1738685534.217710037,0
        eval.go:2172: in MaybeWrapError()
[16:12:19] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges with config 3node-tenant: 69 tests, 1 failures
    logic.go:4231: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:312: error while processing
    logic.go:4231: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:312: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantLogic_synthetic_privileges1662421058
--- FAIL: TestTenantLogic_synthetic_privileges (10.34s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=6</code>
 - <code>shard=38</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #140011 pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogic_synthetic_privileges failed [C-test-failure O-robot P-2 T-sql-foundations X-duplicate branch-release-24.3.4-rc]
- #139890 pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogic_synthetic_privileges failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_synthetic_privileges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_synthetic_privileges [failed](https://tanzanite.cluster.engflow.com/invocations/default/a0d12bc7-c931-44ff-88e0-f1f7962f3021?testReportRun=12&testReportShard=38&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy8zbm9kZS10ZW5hbnQ6M25vZGUtdGVuYW50X3Rlc3Q=) on release-24.3 @ [754142b1d85d078c7e1c4f67539ba1e78d413234](https://github.com/cockroachdb/cockroach/commits/754142b1d85d078c7e1c4f67539ba1e78d413234):


```
=== RUN   TestTenantLogic_synthetic_privileges
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestTenantLogic_synthetic_privileges2553053861
    test_log_scope.go:76: use -show-logs to present logs inline
[15:23:54] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 60 statements
    logic.go:3045: 
         
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:307: SELECT has_table_privilege('testuser4', 'crdb_internal.feature_usage', 'SELECT')
        expected success, but found
        (40001) restart transaction: has_table_privilege(): TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_ABORT_SPAN): "sql txn" meta={id=4300476d key=/Tenant/10/Table/52/1/"testuser4"/"/vtable/crdb_internal/feature_usage" iso=Serializable pri=100.00000000 epo=0 ts=1738769035.201539161,0 min=1738769035.201539161,0 seq=4} lock=true stat=ABORTED rts=1738769035.201539161,0 wto=false gul=1738769035.701539161,0
        eval.go:2172: in MaybeWrapError()
[15:24:01] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 69 statements
[15:24:01] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges with config 3node-tenant: 69 tests, 1 failures
    logic.go:4231: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:312: error while processing
    logic.go:4231: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:312: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantLogic_synthetic_privileges2553053861
--- FAIL: TestTenantLogic_synthetic_privileges (10.44s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=12</code>
 - <code>shard=38</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #140011 pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogic_synthetic_privileges failed [C-test-failure O-robot P-2 T-sql-foundations X-duplicate branch-release-24.3.4-rc]
- #139890 pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogic_synthetic_privileges failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_synthetic_privileges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/logictest/BUILD.bazel
+++ b/pkg/sql/logictest/BUILD.bazel
@@ -62,6 +62,7 @@ go_library(
         "//pkg/kv/kvclient/rangefeed",
         "//pkg/kv/kvserver",
         "//pkg/kv/kvserver/kvserverbase",
+        "//pkg/kv/kvserver/txnwait",
         "//pkg/multitenant/tenantcapabilities",
         "//pkg/security/username",
         "//pkg/server",

--- a/pkg/sql/logictest/logic.go
+++ b/pkg/sql/logictest/logic.go
@@ -41,6 +41,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvserverbase"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/txnwait"
 	"github.com/cockroachdb/cockroach/pkg/multitenant/tenantcapabilities"
 	"github.com/cockroachdb/cockroach/pkg/security/username"
 	"github.com/cockroachdb/cockroach/pkg/server"
@@ -1707,6 +1708,21 @@ func (t *logicTest) newCluster(
 				if _, err := conn.Exec(query, tenantID.ToUint64(), value); err != nil {
 					t.Fatal(err)
 				}
+			}
+			if _, err := conn.Exec(
+				"RESET CLUSTER SETTING kv.closed_timestamp.target_duration",
+			); err != nil {
+				t.Fatal(err)
+			}
+			if _, err := conn.Exec(
+				"RESET CLUSTER SETTING kv.closed_timestamp.side_transport_interval",
+			); err != nil {
+				t.Fatal(err)
+			}
+			if _, err := conn.Exec(
+				"RESET CLUSTER SETTING kv.rangefeed.closed_timestamp_refresh_interval",
+			); err != nil {
+				t.Fatal(err)
 			}
 		}
 
@@ -4372,6 +4388,18 @@ func RunLogicTest(
 	}
 	if *printErrorSummary {
 		defer lt.printErrorSummary()
+	}
+	if config.UseSecondaryTenant == logictestbase.Always {
+		// Under multitenant configs running in EngFlow, we have seen that logic
+		// tests can be flaky due to an overload condition where schema change
+		// transactions do not heartbeat quickly enough. This allows background jobs
+		// such as the spanconfig reconciler or the job registry "remove claims from
+		// dead sessions" loop.
+		// See https://github.com/cockroachdb/cockroach/pull/140400#issuecomment-2634346278
+		// and https://github.com/cockroachdb/cockroach/issues/140494#issuecomment-2640208187
+		// for a detailed analysis of this issue.
+		cleanup := txnwait.TestingOverrideTxnLivenessThreshold(30 * time.Second)
+		defer cleanup()
 	}
 	// Each test needs a copy because of Parallel
 	serverArgsCopy := serverArgs

